### PR TITLE
test: use declarator to enable large payload response

### PR DESCRIPTION
**Pull request by yugangw-msft**
//CC: @tjprescott  @troydai @lmazuel   
Another way to support scenario tests playback with large response
- I keep the #5568 open so we can compare the pros and cons
- The current [LargeResponseProcessor](https://github.com/Azure/azure-python-devtools/blob/master/src/azure_devtools/scenario_tests/recording_processors.py#L86) only works for non-json payload, which is why storage test works, but every json payload based test will NOT work as the tweaking causes invalid json content. Hence I suggest, for big json payload, we just raise in the LargeResponseProcessor and tell users to use the new flag to opt-in big size, for other types we will still tweak like we are doing now, so storage tests continue work with shrunken recordings 

Since this has no end user impacts, I don't have strong opinions towards either options, so just tell me which one you prefer 

### General Guidelines

- [na] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [na] Each command and parameter has a meaningful description.
- [na] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5588
This is an **experimental preview** for @microsoft users.

File: src/azure-cli-testsdk/azure/cli/testsdk/preparers.py [hunk 1]
```diff
@@ -170,6 +170,18 @@ def remove_resource(self, name, **kwargs):
         if not self.dev_setting_sp_name:
             execute(self.cli_ctx, 'az ad sp delete --id {}'.format(self.result['appId']))
 
+# Function wise, enabling large payload recording has nothing to do with resource preparers 
+# We still base on it so that this decorator can chain with other preparers w/o too much hacks
+class LargeResponsePreparer(AbstractPreparer):
```
Comment (tjprescott) [inline src/azure-cli-testsdk/azure/cli/testsdk/preparers.py]: Preparer is really meant for resources. I agree with your logic for using the AbstractPreparer, but I'd recommend a name like "AllowLargeResponse".

**Comment by tjprescott** [comment]
Fixes #4127.

**Comment by tjprescott** [comment]
Actually... one more comment.  This should really be in devtools.  If you plan on making a change in devtools that fails the recording if a large response body is encountered with JSON... then it can't direct them to a fix that only exists in the CLI.  That fix would also need to be in devtools.

File: src/azure-cli-testsdk/azure/cli/testsdk/preparers.py [hunk 2]
```diff
@@ -170,6 +170,18 @@ def remove_resource(self, name, **kwargs):
...
+    def __init__(self, size_kb=1024):
+        self.size_kb = size_kb
+        super(LargeResponsePreparer, self).__init__('nanana', 20)
```
Comment (lmazuel) [inline src/azure-cli-testsdk/azure/cli/testsdk/preparers.py]: nanana?

Comment (yugangw-msft) [reply src/azure-cli-testsdk/azure/cli/testsdk/preparers.py]: 3 `n/a` to make the `AbstractPreparer` happy on the random name length. :)

Comment (yugangw-msft) [reply src/azure-cli-testsdk/azure/cli/testsdk/preparers.py]: it is not really used, but a workaround to get in the decorator's chain

**Comment by yugangw-msft** [comment]
Close it and i am moving the change to devtools https://github.com/Azure/azure-python-devtools/pull/38

**Pull request closed without merging**